### PR TITLE
[dev] add TOOD

### PR DESCRIPTION
**Pull request by ghostxsl**
- 新增TOOD模型
- 新增tood_head, assigners, atss_assigner, task_aligned_assigner等模块
- tood_r50_fpn_1x_coco模型复现精度为42.8

### CODE REVIEW

**Review by wangxinxin08 — DISMISSED** [review]
LGTM

**Review by lyuwenyu — DISMISSED** [review]

File: configs/tood/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# TOOD
+
+## Introduction
+
+
+TOOD is an object detection model. We reproduced the model of the paper.
```
Comment (qingqing01) [inline configs/tood/README.md:7]: TOOD增加链接

**Review by qingqing01 — COMMENTED** [review]
use pre-commit to format code

File: configs/tood/README.md (hunk 1 above)
Comment (ghostxsl) [reply configs/tood/README.md:7]: done, thx~

File: ppdet/modeling/heads/tood_head.py [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+        self._init_weights()
+
+    def _init_weights(self):
+        normal_(self.conv.weight, std=0.001)
+
+    def forward(self, inputs):
+        out = self.conv(inputs)
+        out = self.norm(out)
+        return out
```
Comment (qingqing01) [inline ppdet/modeling/heads/tood_head.py]: https://github.com/PaddlePaddle/PaddleDetection/blob/5b949596ea7603cd79e3fc9067766bbc79a3e93d/ppdet/modeling/layers.py#L111 

能复用这个嘛？

Comment (ghostxsl) [reply ppdet/modeling/heads/tood_head.py]: done

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**